src/utils/etherspot.js

```diff
@@ -204,31 +204,12 @@ export const buildExchangeOffer = (
   return { provider, fromAsset, toAsset, fromAmount, toAmount, exchangeRate: 1 / exchangeRate, transactions };
 };
 
-export const prepareAssets = (assets: Asset[]): Asset[] => {
-  let result = assets;
-  result = appendEthIfNeeded(result);
-  result = orderBy(result, [(asset) => (asset.iconUrl ? 0 : 1)]);
-  result = uniqBy(result, (asset) => asset.address.toUpperCase());
+export const appendNativeAssetIfNeeded = (chain: Chain, assets: Asset[]): Asset[] => {
+  const nativeAsset = nativeAssetPerChain[chain];
 
-  return result;
-};
-
-const appendEthIfNeeded = (assets: Asset[]): Asset[] => {
-  const hasEth = assets.some(({ symbol }) => symbol === ETH);
-  if (hasEth) return assets;
-
-  // eslint-disable-next-line i18next/no-literal-string
-  const iconUrl = 'https://tokens.1inch.exchange/0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee.png';
-  const eth = {
-    address: EthersConstants.AddressZero,
-    name: 'Ethereum', // eslint-disable-line i18next/no-literal-string
-    symbol: ETH,
-    decimals: 18,
-    iconUrl,
-    iconMonoUrl: iconUrl,
-  };
-
-  return [eth, ...assets];
+  // TODO: Switch to zero address check when supported assets per chain are implemented.
```

Comment:
This part should to be addresses in this PR or supported assets per chains (whichever merges last).